### PR TITLE
Renam Github workflow job names

### PR DESCRIPTION
**Pull request by sven1103**
This PR provides declarative names for the workflow jobs, so we can use them for branch protection rules.

### CODE REVIEW

File: .github/workflows/nexus-publish-snapshots.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
       - development
 
 jobs:
-  build:
+  publish_snapshot:
```
Comment (KochTobi) [inline .github/workflows/nexus-publish-snapshots.yml:13]: NIT: we named this `publish_snapshots` in sync with the GitHub action name. I am okay with both

**Review by KochTobi — APPROVED** [review]

**Review by KochTobi — CHANGES_REQUESTED** [review]
codeql is missing

**Review by KochTobi — COMMENTED** [review]
We renamed the `analyze` check in the other repositories. I would prefer to have it the same for required checks across repositories

File: .github/workflows/nexus-publish-snapshots.yml (hunk 1 above)
Comment (sven1103) [reply .github/workflows/nexus-publish-snapshots.yml:13]: But its technically wrong, it deploys only ONE snapshot :)

**Comment by sven1103** [comment]
CodeQL was already named properly

**Review by KochTobi — APPROVED** [review]

File: .github/workflows/nexus-publish-snapshots.yml (hunk 1 above)
Comment (KochTobi) [reply .github/workflows/nexus-publish-snapshots.yml:13]: yes :) the action files name contains the plural tough 🤷‍♂️ As we do not actively use this I think it is fine

Comment (KochTobi) [reply .github/workflows/nexus-publish-snapshots.yml:13]: yes :) the action files name contains the plural tough 🤷‍♂️ As we do not actively use this I think it is fine

**Pull request merged**